File: Mock_Project_1/SmartShop/src/main/java/vn/itsol/smartshop/controller/CategoryController.java
package vn.itsol.smartshop.controller;

import java.util.List;

import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import vn.itsol.smartshop.dao.CategoryDao;
import vn.itsol.smartshop.dao.impl.CategoryDaoImpl;
import vn.itsol.smartshop.model.Category;

@RestController
@RequestMapping("/category")
public class CategoryController {
	
	CategoryDao dao=new CategoryDaoImpl();
	
	@RequestMapping(value="/getall",method=RequestMethod.GET,produces="application/json")
	public List<Category> getAll(){
		return dao.getAll();
	}
	
}

File: Mock_Project_1/SmartShop/src/main/webapp/js/angular/category.controller.js
app.controller('CategoryController', [ 'async', function(async) {
	var self = this;
	self.items = async;
} ]);

app.controller('getCate', [ '$scope', '$http', function($scope, $http) {
	$http({
		method : 'GET',
		url : 'http://localhost:8080/SmartShop/category/getall'
	}).then(function(response) {
		$scope.categories = response.data;
	});
} ]);